src/VestingEarndrop/VestingEarndrop.sol

```diff
@@ -201,21 +241,69 @@ contract VestingEarndrop is Ownable2Step, EIP712 {
       revert Unauthorized();
     }
 
+    earndrop.pendingRevoke = true;
+    emit EarndropRevokeRequested(earndropId, msg.sender);
+  }
+
+  /**
+   * @dev Cancels a pending revoke request for an Earndrop.
+   * @param earndropId The unique ID of the Earndrop.
+   */
+  function cancelRevokeEarndrop(uint256 earndropId) external {
+    Earndrop storage earndrop = earndrops[earndropId];
+    if (earndrop.earndropId == 0) {
+      revert InvalidParameter("Earndrop does not exist");
+    }
+    if (!earndrop.pendingRevoke) {
+      revert InvalidParameter("No pending revoke request");
+    }
+    if (msg.sender != earndrop.admin) {
+      revert Unauthorized();
+    }
+
+    earndrop.pendingRevoke = false;
+
+    emit EarndropRevokeCancelled(earndropId, msg.sender);
+  }
+
+  /**
+   * @dev Confirms the activation of an Earndrop by transferring funds.
+   * @param earndropId The unique ID of the Earndrop.
+   */
+  function confirmRevokeEarndrop(uint256 earndropId) external onlyOwner {
+    Earndrop storage earndrop = earndrops[earndropId];
+    if (earndrop.earndropId == 0) {
+      revert InvalidParameter("Earndrop does not exist");
+    }
+    if (earndrop.revoked) {
+      revert InvalidParameter("Earndrop already revoked");
+    }
+    if (!earndrop.pendingRevoke) {
+      revert InvalidParameter("No pending revoke request");
+    }
+    if (!earndrop.confirmed) {
+      revert InvalidParameter("Earndrop not confirmed");
+    }
+
```

Comment:
revoke Earndrop doesn't need to this complicated. Just do it this way:

1. Owner can set a flag indicating if an earndrop as revocable or not.
2. Earndrop admin can only revoke revocable earndrop.